### PR TITLE
TkDQM: Add DQM modules for monitoring VectorHits

### PR DESCRIPTION
**Pull request by sroychow**
#### PR description:
Work done by @gsaha009 to introduce DQM modules to monitor VectorHits. Two collections of vector hits are monitored - ```accepted``` and ```rejected```. 

#### PR validation:
Wf ```20834.9``` runs fine.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported 
NA

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * workflow = 20834.9

**Comment by mmusich** [comment]
please test

File: DQM/SiTrackerPhase2/plugins/Phase2OTMonitorVectorHits.cc [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+    psd0.add<std::string>("name", "CurvatureVsEtaProf_S");
+    psd0.add<std::string>("title", "Curvature vs #eta (strip);#eta ;curvature ;");
+    psd0.add<int>("NxBins", 25);
+    psd0.add<double>("xmin", -5.0);
+    psd0.add<double>("xmax", 5.0);
+    psd0.add<double>("ymin", -0.05);
+    psd0.add<double>("ymax", 0.05);
+    psd0.add<bool>("switch", true);
+    desc.add<edm::ParameterSetDescription>("CurvatureVsEta_S", psd0);
+  }
```
Comment (mmusich) [inline DQM/SiTrackerPhase2/plugins/Phase2OTMonitorVectorHits.cc:459]: I am missing plots for some of the quantities one can observe, namely:

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/TrackerRecHit2D/interface/VectorHit.h#L81

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/TrackerRecHit2D/interface/VectorHit.h#L82

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/TrackerRecHit2D/interface/VectorHit.h#L82

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/TrackerRecHit2D/interface/VectorHit.h#L88

File: DQM/SiTrackerPhase2/plugins/Phase2OTMonitorVectorHits.cc [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+      const float curverr = vechitIt->curvatureError();
+      const float eta = globalVec.eta();
+      const float phi = globalVec.phi();
+      if (mType == TrackerGeometry::ModuleType::Ph2PSP) {
+        globalXY_P_->Fill(gx, gy);
+        globalRZ_P_->Fill(gz, gr);
+        //layer wise histo
+        layerMEs_[key].localPosXY_P->Fill(lp.x(), lp.y());
+        layerMEs_[key].curvature_P->Fill(curvature);
+        if (curvature != 0.f)
```
Comment (mmusich) [inline DQM/SiTrackerPhase2/plugins/Phase2OTMonitorVectorHits.cc:169]: should we log a failure here?

File: DQM/SiTrackerPhase2/plugins/Phase2OTMonitorVectorHits.cc [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+        layerMEs_[key].phi_P->Fill(phi);
+        layerMEs_[key].eta_P->Fill(eta);
+        layerMEs_[key].curvatureVsEta_P->Fill(eta, curvature);
+      } else if (mType == TrackerGeometry::ModuleType::Ph2SS) {
+        globalXY_S_->Fill(gx, gy);
+        globalRZ_S_->Fill(gz, gr);
+        //layer wise histo
+        layerMEs_[key].localPosXY_S->Fill(lp.x(), lp.y());
+        layerMEs_[key].curvature_S->Fill(curvature);
+        if (curvature != 0.f)
```
Comment (mmusich) [inline DQM/SiTrackerPhase2/plugins/Phase2OTMonitorVectorHits.cc:180]: as above?

File: DQM/SiTrackerPhase2/plugins/Phase2OTMonitorVectorHits.cc (hunk 1 above)
Comment (sroychow) [reply DQM/SiTrackerPhase2/plugins/Phase2OTMonitorVectorHits.cc:459]: Thanks! Yes chi2, momentum can be added. The dimension is always 4, that's why it has not been added.

Comment (mmusich) [reply DQM/SiTrackerPhase2/plugins/Phase2OTMonitorVectorHits.cc:459]: > The dimension is always 4, that's why it has not been added.

mmmh, indeed

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/TrackerRecHit2D/interface/VectorHit.h#L139

wondering what purpose it serves...

File: DQM/SiTrackerPhase2/python/Phase2TrackerDQMFirstStep_cff.py [hunk 4]
```diff
@@ -3,10 +3,23 @@
...
+
+from Configuration.ProcessModifiers.vectorHits_cff import vectorHits
+trackerphase2DQMSource_wVhits = cms.Sequence( pixDigiMon
+                                       + otDigiMon
+                                       +rechitMonitorIT
+                                       + clusterMonitorIT
+                                       + clusterMonitorOT
+                                       + acceptedVecHitsmon
+                                       + rejectedVecHitsmon
+)
```
Comment (mmusich) [inline DQM/SiTrackerPhase2/python/Phase2TrackerDQMFirstStep_cff.py:23]:
```suggestion
trackerphase2DQMSource_wVhits = trackerphase2DQMSource.copy()
trackerphase2DQMSource_wVhits += cms.Sequence(acceptedVecHitsmon +rejectedVecHitsmon)
```

is more compact and should be equivalent.

**Comment by mmusich** [comment]
please test

**Comment by emanueleusai** [comment]
This adds about 110MB of histogram memory, about 1090 monitoring elements. Are they all necessary?

**Comment by mmusich** [comment]
@emanueleusai 

> This adds about 110MB of histogram memory, about 1090 monitoring elements. Are they all necessary?

These histograms are added only in the workflows running the vector hits reconstruction (a very limited subset of the full matrix, activated by the `seedingDeepCore` modifier.
They are needed now during the development/commissioning phase and will be removed/merged with the regular OT monitoring once that is over.

**Comment by emanueleusai** [comment]
@mmusich thank you very much for the clarification

**Comment by emanueleusai** [comment]
+1

**Comment by rappoccio** [comment]
+1

**Pull request merged**